Sources/GRPCHTTP2TransportNIOPosix/HTTP2ServerTransport+Posix.swift

```diff
@@ -62,6 +158,25 @@ extension HTTP2ServerTransport {
           }
         }
 
+      self.listeningAddressState.withLockedValue { state in
+        switch state {
+        case .idle(let idleState):
+          if let address = serverChannel.channel.localAddress {
+            idleState.listeningAddressPromise.succeed(address)
+          } else {
+            let failedToGetAddressError = RuntimeError(
+              code: .transportError,
+              message: "An address is not available."
+            )
+            idleState.listeningAddressPromise.fail(failedToGetAddressError)
```

Comment:
I _think_ this can be `nil` for `vsock`, in which case we should return `self.address`

Reply:
Interesting - should we always fall back to returning `self.address` if this is `nil` (i.e. even if it's not a `vsock`)?

Reply:
Pfft, good question. 

I think we should only succeed the promise if it's a `vsock` and fail the promise otherwise. In the failure case we should also add an assertion failure (because we might add a new address type in the future, and that'll help us surface where we should be handling it)